src/Manager.php

```diff
@@ -157,8 +158,12 @@ protected function getFilesFromPaths(array $paths, array $extensions, array $exc
                 $files[] = $path;
             } elseif (is_dir($path)) {
                 $iterator = new \RecursiveDirectoryIterator($path, \FilesystemIterator::SKIP_DOTS);
-                if (!empty($excluded)) {
-                    $iterator = new RecursiveDirectoryFilterIterator($iterator, $excluded);
+                if (!empty($excluded) && !empty($OnlyFilesChanged)) {
+                    $iterator = new RecursiveDirectoryFilterIterator($iterator, $excluded, $OnlyFilesChanged);
+                } elseif (!empty($OnlyFilesChanged) ) {
+                    $iterator = new RecursiveDirectoryFilterIterator($iterator, $excluded, $OnlyFilesChanged);
+                } elseif (!empty($excluded)) {
+                    $iterator = new RecursiveDirectoryFilterIterator($iterator, $excluded, $OnlyFilesChanged);
```

Comment:
What about if you get `$onlyFilesChanged` array then create iterator only with those files (and don't use files from `$iterator`)?